storage.go

```diff
@@ -235,79 +162,54 @@ func (s *Storage) Boot(ctx context.Context) error {
 	return nil
 }
 
-func (s *Storage) Put(ctx context.Context, kv microstorage.KV) error {
-	key, value := kv.Key(), kv.Val()
+func (s *Storage) Delete(ctx context.Context, k microstorage.K) error {
+	var err error
 
 	var body []byte
 	{
-		v := struct {
-			Data map[string]string `json:"data"`
-		}{
-			Data: map[string]string{
-				key: value,
-			},
-		}
+		v := storageConfigJSONPatch{}
+		v.Spec.Storage.Data[k.Key()] = nil
 
-		var err error
 		body, err = json.Marshal(&v)
 		if err != nil {
-			return microerror.Maskf(err, "marshaling %#v", v)
+			return microerror.Mask(err)
```

Comment:
Why you removed annotatoins from errors? I think they are pretty valuable.

Reply:
They where annoying me while developing and getting it right in my head. They do not provide any additional value to me personally. What I need is the actual message which is already provided by the given error and its call stack.

Reply:
You say you like when things don't make you think. Every additional string here makes me think and scanning the error to get to the bottom of its cause just makes me think and slows me down.

Reply:
I think masking is useful when you produce an error. When you only forward it additional masking does not provide anything for me personally.

Reply:
I don't want to think too much on unimportant stuff, but error handling is important for me. I don't think stack traces are reliable. Codebase changes. I also can't wrap my head around why masking is only useful when you produce error. What additional value `not found` brings? I think masking only when you produce error isn't useful at all. Then we have problems like that: https://github.com/giantswarm/opsctl/issues/110

We ignore importance of error messages and this makes me sad.

Disagree and commit.

Reply:
I feel like we did not understand each other here. Just some examples below to make my point more clear. Thanks for approving so I can go ahead. We can foster our error handling and agreement down the road. We can also write something more down in our `fmt` repo when things are unclear. 

I think this is good. There is an error given so nothing has to be done. No masking is required because the received error should contain all info that it needs. We only mask for the call stack.

```
if err != nil {
	return microerror.Mask(err)
}
```

I think this is good. There is no error because the situation produces an unexpected behaviour based on other state. So we make up an error. `not found` itself is not useful, so masking it and annotating it with additional information makes total sense and helps. 

```
if !ok {
	return microerror.Maskf(notFoundError, "cluster ID '%s', clusterID)
}
```

I think this is odd and does not help. 

```
if err != nil {
	return microerror.Maskf(err, "something went wrong and I talk even though I have no clue, oh by the way here is some random variable that was involved in producing the issue somehow '%s'', someThing)
}
```

Reply:
Counter example:

```diff
-				return microerror.Maskf(err, "creating TPO %#v", tpo)
+				return microerror.Mask(err)
```

With message like `wrong status: SOME_NUMBER` this is worth nothing even with up to date stack trace. But with the actual object content and `creating` word (as opposed to long message you made up). You can try `kubectl create object_content.yaml` and debug even w/o stack trace.